processor/ratelimitprocessor/gubernator.go

```diff
@@ -75,21 +95,34 @@ func newGubernatorDaemonConfig(logger *zap.Logger) (gubernator.DaemonConfig, err
 	return conf, nil
 }
 
-func newGubernatorRateLimiter(cfg *Config, set processor.Settings) (*gubernatorRateLimiter, error) {
-	daemonCfg, err := newGubernatorDaemonConfig(set.Logger)
+func newGubernatorRateLimiter(cfg *Config, logger *zap.Logger, telemetryBuilder *metadata.TelemetryBuilder) (*gubernatorRateLimiter, error) {
+	daemonCfg, err := newGubernatorDaemonConfig(logger)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create gubernator daemon config: %w", err)
 	}
 
 	return &gubernatorRateLimiter{
-		cfg:       cfg,
-		set:       set,
-		behavior:  gubernator.Behavior_BATCHING,
-		daemonCfg: daemonCfg,
+		cfg:              cfg,
+		logger:           logger,
+		behavior:         gubernator.Behavior_BATCHING,
+		daemonCfg:        daemonCfg,
+		telemetryBuilder: telemetryBuilder,
+		classResolver:    noopResolver{},
 	}, nil
 }
 
-func (r *gubernatorRateLimiter) Start(ctx context.Context, _ component.Host) (err error) {
+func (r *gubernatorRateLimiter) Start(ctx context.Context, host component.Host) (err error) {
+	if res := r.cfg.ClassResolver; res.String() != "" {
```

Comment:
nit: better to detect for empty extension id in config validation?

Reply:
I meant to explicitly check if resolver is defined, then it cant be empty.

Reply:
Added config validation so there's an error if classes are set (they are optional), but no class_resolver is set.